fastats/maths/ols.py

```diff
@@ -32,6 +33,42 @@ def ols_qr(A, b):
     return solve_triangular(R, Q.T @ b)
 
 
+def ols_cholesky(A, b):
+    """
+    Ordinary Least-Squares Regression Coefficients
+    Estimation.
+
+    If (A.T @ A) @ x = A.T @ b and A is full rank
+    then there exists an upper triangular matrix
+    R such that:
+
+    (R.T @ R) @ x = A.T @ b
+    R.T @ w = A.T @ b
+    R @ x = w
+
+    Find R using Cholesky decomposition.
+    """
+    R = cholesky(A.T @ A)
+    w = solve_triangular(R, A.T @ b, trans='T')
+    return solve_triangular(R, w)
```

Comment:
Brilliant stuff!

Reply:
But it won't JIT in the current form, right? `solve_triangular` doesn't seem to be supported in numba: http://numba.pydata.org/numba-doc/0.35.0/reference/numpysupported.html

Reply:
Indeed, these won't JIT (same for the existing ols_qr).  We'd need to swap out the factorisation methods with our own at some point to get it to play nicely.